### PR TITLE
update test-requirements.txt to match fixed-requirements.txt

### PR DESCRIPTION
**Pull request by cognifloyd**
- Removes two out-of-date requirements from fixed-requirements.txt that are no longer needed anywhere except for in tests.
- Harmonizes remaining differences by updating test-requirements.txt

### CODE REVIEW

**Comment by cognifloyd** [comment]
The `Orquesta CI / Skip Duplicate Jobs Pre Job (pull_request)` job is skipping the latest commit. :frowning_face:
Here is the previous successful run that it is detecting: https://github.com/StackStorm/st2/actions/runs/1614687831

**Review by winem — APPROVED** [review]
Very good change! Looks like the logic that was in place regarding the keys and restored keys had a few caveats anyway.

File: test-requirements.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ pre-commit==2.1.0
 bandit==1.7.0
 ipython<6.0.0
 isort>=4.2.5
-mock==3.0.3
+mock==4.0.3
```
Comment (cognifloyd) [inline test-requirements.txt:13]: cf
https://github.com/StackStorm/st2/blob/6cd582e65f3ef8d7b1ad7c7b3df47e632de9f67b/fixed-requirements.txt#L71

File: test-requirements.txt [hunk 2]
```diff
@@ -22,10 +22,10 @@ nose-timer==1.0.1
 # splitting tests run on a separate CI machines
 nose-parallel==0.4.0
 # Required by st2client tests
-pyyaml==5.4
+pyyaml==5.4.1
```
Comment (cognifloyd) [inline test-requirements.txt:25]: https://github.com/StackStorm/st2/blob/6cd582e65f3ef8d7b1ad7c7b3df47e632de9f67b/fixed-requirements.txt#L48

File: test-requirements.txt [hunk 3]
```diff
@@ -22,10 +22,10 @@ nose-timer==1.0.1
...
 RandomWords
-gunicorn==19.9.0
-psutil==5.6.6
+gunicorn==20.1.0
```
Comment (cognifloyd) [inline test-requirements.txt:27]: https://github.com/StackStorm/st2/blob/6cd582e65f3ef8d7b1ad7c7b3df47e632de9f67b/fixed-requirements.txt#L20

File: test-requirements.txt [hunk 4]
```diff
@@ -22,10 +22,10 @@ nose-timer==1.0.1
...
+psutil==5.8.0
```
Comment (cognifloyd) [inline test-requirements.txt:28]: https://github.com/StackStorm/st2/blob/6cd582e65f3ef8d7b1ad7c7b3df47e632de9f67b/fixed-requirements.txt#L74

File: fixed-requirements.txt [hunk 5]
```diff
@@ -35,7 +35,6 @@ oslo.config>=1.12.1,<1.13
 oslo.utils<5.0,>=4.0.0
 paramiko==2.7.2
 passlib==1.7.4
-prance==0.9.0
```
Comment (cognifloyd) [inline fixed-requirements.txt:38]: cf
https://github.com/StackStorm/st2/blob/6cd582e65f3ef8d7b1ad7c7b3df47e632de9f67b/test-requirements.txt#L33

File: fixed-requirements.txt [hunk 6]
```diff
@@ -74,6 +73,5 @@ nose-parallel==0.4.0
 psutil==5.8.0
 python-dateutil==2.8.1
 python-statsd==2.1.0
-ujson==1.35
```
Comment (cognifloyd) [inline fixed-requirements.txt:77]: ujson is only used for benchmark tests now. We use orjson everywhere else.
https://github.com/StackStorm/st2/blob/6cd582e65f3ef8d7b1ad7c7b3df47e632de9f67b/test-requirements.txt#L45

**Review by cognifloyd — COMMENTED** [review]
To simplify comparison, here are some comments to show the version in the other file.

**Comment by cognifloyd** [comment]
I asked for a review from a senior maintainer because we need a repo admin to merge this to work around the skip duplicate jobs issue which is preventing required statuses from being reported.

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -428,6 +428,7 @@ jobs:
 
       # GitHub is juggling how to set vars for multiple shells. Protect our PATH assumptions.
       PATH: /home/runner/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
```
Comment (cognifloyd) [inline .github/workflows/ci.yaml:431]: Add a newline to force CI to rerun and get around the "skip duplicates" jobs.

**Pull request merged**